proposals/devcontainer-templates.md

```diff
@@ -0,0 +1,169 @@
+# Dev Container Templates reference
+
+Development container "Templates" are self-contained, shareable units of development container configuration. The name comes from the idea that adding a Template to your project helps it get up and running with a containerized environment. The Templates describe the appropriate container image, [Features](https://github.com/devcontainers/spec/blob/main/proposals/devcontainer-features.md), runtime arguments for starting the container, and VS Code extensions that should be installed. Each provides a container configuration file ([`.devcontainer.json`](/docs/specs/devcontainer-reference.md#devcontainerjson)) and other needed files that you can drop into any existing folder as a starting point for containerizing your project.
+
+Template metadata is captured by a `devcontainer-template.json` file in the root folder of the Template.
+
+## Folder structure 
+
+A Template is a self contained entity in a folder with at least a `devcontainer-template.json` and [`devcontainer.json`](/docs/specs/devcontainer-reference.md#devcontainerjson).  Additional files are permitted and are packaged along side the required files.
+
+```
++-- template
+|    +-- devcontainer-template.json
+|    +-- .devcontainer.json
+|    +-- (other files)
+```
+
+## devcontainer-template.json properties
```

Comment:
@Chuxel  For the `publisher` info which is needed by the UI -  I believe we will have a similar index for Templates as we have for [Features](https://github.com/devcontainers/devcontainers.github.io/blob/gh-pages/_data/collection-index.yml#L2) and this index will be used by the tooling to display list of Templates. We can then make use of `maintainer` property to display to `publisher/maintainer` name wherever needed.  

 Adding `publisher` prop to `devcontainer-template.json` feels a little unsafe as any author can claim to be someone else and we currently don't have identity checking mechanism in place. At least, with the search index, we will manually look at the PR created by the author (to add his Templates to crawler) and confirm if the maintainer prop matches with  other details.

Reply:
Agreed, added `publisher` 👍